### ISSUE TITLE
Document the default behavior of `require-error` and `fn-pattern`

### ISSUE THREAD

**Issue by ian-h-chamberlain** [Antonboom/testifylint#44]
From reading https://github.com/Antonboom/testifylint#require-error it's not immediately clear to me _which_ calls will be flagged. Initially I assumed there was some default regex for `fn-pattern` and the user could override it by specifying configuration.

However, it looks like there is actually a hardcoded list of function names, and the regex only serves to filter them down further: https://github.com/Antonboom/testifylint/blob/fc4c8235262c71530c5870869ebe8baa4ad44e6a/internal/checkers/require_error.go#L121C6-L121C7

1. Is my understanding of the configuration accurate?
2. If so, would you consider adding some documentation to the README explaining this?

Thanks!

**Comment by Antonboom** [comment]
Hi!

> the user could override it by specifying configuration

exactly so

> and the regex only serves to filter them down further

this is right too

in other words – `require-error` marks any error assertions (how README said), but you could select the specific assertions by `fn-pattern`

example [here](https://github.com/golangci/golangci-lint/blob/84442f26446b0fa27265b075185bb481075e1c95/.golangci.reference.yml#L2023):
```yml
    require-error:
      # Regexp for assertions to analyze. If defined then only matched assertions will be reported.
      # Default: ""
      fn-pattern: ^(Errorf?|NoErrorf?)$
```

^ it means `require` will be required (sorry for tautology 🙂) only for 

```go
assert.NoError(t, err)
assert.NoErrorf(t, err, "msg")
assert.Error(t, err)
assert.Errorf(t, err, "msg")
```

other errors assertions (from hardcoded list) will be skipped.

You could not specify `require-error` for assertions like `Negative` or `WithinDuration` or any other not-error-assertion fn-pattern, because it doesn't make sense.

---

Please provide an example and goal what you want to reach, thanks

**Comment by ian-h-chamberlain** [comment]
Got it, thanks for explaining in more detail. I think to make it more clear perhaps I'd suggest a change like this:

> Also you can configure functions to analyze via `--require-error.fn-pattern` flag.

into something like

> By default, all `*Error*`-related functions will be checked (e.g. `assert.ErrorContains`, `assert.NoError`, `assert.ErrorAs`).
>
> You can configure `--require-error.fn-pattern` to limit the check to certain calls, for example:
> `--require-error.fn-pattern='^(Errorf?|NoErrorf?)$'`
> would only check `Error`, `Errorf`, `NoError`, and `NoErrorf`.

I think that would make it more clear that not _every_ `assert.*` function call will be checked, but just the `Error` related ones (i.e. the hardcoded list). It probably doesn't make sense to put the entire list in the README but some examples would be helpful, I think.

Thanks again!
